src/main/java/org/Lab1/Operation.java

```diff
@@ -0,0 +1,190 @@
+package org.Lab1;
+
+import java.util.Objects;
+import java.util.Scanner;
+
+public class Operation {
+    public static void movieListViewer(Movie[] moviesList, Scanner sc) {
+        if (moviesList.length == 0) {
+            System.out.println("No movies to display.");
+        } else {
+            System.out.println("\nMovies and Ratings:");
+            for (Movie m : moviesList) {
+                System.out.println(m.toString());
+            }
+        }
+    }
+    public static Movie[] movieAdder(Movie[] moviesList, Scanner sc) {
+        System.out.println("Enter movie:");
+        String nameOfMovie = sc.nextLine();
+        System.out.println("Enter rating:");
+        double ratingOfMovie = sc.nextDouble();
+        sc.nextLine(); // Consume leftover newline
+
+
+        Movie[] newArray = new Movie[moviesList.length + 1];
+        System.arraycopy(moviesList, 0, newArray, 0, moviesList.length);
+        newArray[moviesList.length] = new Movie(nameOfMovie, ratingOfMovie);
+        return newArray;
+
+    }
+    public static Movie[] sortMovies(Movie[] moviesList) {
+        int n = moviesList.length;
+
+        for (int i = 0; i < n - 1; i++) {
+            for (int j = 0; j < n - 1 - i; j++) {
+                // Compare ratings for ascending order
+                if (moviesList[j].getRating() > moviesList[j + 1].getRating()) {
+
+                    // Swap movies
+                    Movie temp = moviesList[j];
+                    moviesList[j] = moviesList[j + 1];
+                    moviesList[j + 1] = temp;
+                }
+            }
+        }
+
+        return moviesList;
+    }
+    public static int findMovie(Scanner sc,Movie[] moviesList) {
+        int filmIndex = 0;
+        System.out.println("Enter movie Name");
+        String movieName = sc.nextLine();
+        boolean findMovie = false;
+        for (int z = 0; z < moviesList.length; z++) {
+            filmIndex = z;
+            if (Objects.equals(moviesList[z].getName(), movieName)) {
```

Comment:
you can move this searching logic to another method, which can be helpful to your update and delete functionality, write once use everywhere)